packages/edit-site/src/components/page-templates/template-actions.js

```diff
@@ -34,36 +34,59 @@ export function useResetTemplateAction() {
 			id: 'reset-template',
 			label: __( 'Reset template' ),
 			isPrimary: true,
+			isBulk: true,
 			icon: backup,
 			isEligible: isTemplateRevertable,
-			async callback( template ) {
+			async callback( templates ) {
 				try {
-					await revertTemplate( template, { allowUndo: false } );
-					await saveEditedEntityRecord(
-						'postType',
-						template.type,
-						template.id
+					await Promise.all(
+						templates.map( ( template ) => {
+							return revertTemplate( template, {
+								allowUndo: false,
+							} );
+						} )
+					);
+					await Promise.all(
+						templates.map( ( template ) => {
+							return saveEditedEntityRecord(
+								'postType',
+								template.type,
+								template.id
+							);
+						} )
 					);
 
 					createSuccessNotice(
-						sprintf(
-							/* translators: The template/part's name. */
-							__( '"%s" reverted.' ),
-							decodeEntities( template.title.rendered )
-						),
+						templates.length > 1
+							? sprintf(
+									/* translators: The number of items. */
+									__( '%s items reverted reverted.' ),
+									decodeEntities( templates.length )
+							  )
+							: sprintf(
+									/* translators: The template/part's name. */
```

Comment:
For consistency with following translators comments, I'd suggest to use the same format:
/* translators: %s: The template or template part's name. */